src/terminal/parser/InputStateMachineEngine.cpp

```diff
@@ -751,8 +751,9 @@ DWORD InputStateMachineEngine::_GetGenericKeysModifierState(const VTParameters p
     //   HOME, END, PAGE UP, PAGE DOWN, and direction keys in the clusters to the left
     //   of the keypad; and the divide (/) and ENTER keys in the keypad.
     // This snippet detects the non-direction keys
-    const GenericKeyIdentifiers identifier = parameters.at(0);
-    if (identifier <= GenericKeyIdentifiers::Next)
+    const auto identifier = parameters.at(0).value();
+    if (identifier >= WI_EnumValue(GenericKeyIdentifiers::GenericHome) &&
+        identifier <= WI_EnumValue(GenericKeyIdentifiers::Next))

```

Comment:
I don't see how this resolves the reported issue. We aren't dealing with a key that produces a zero or omitted parameter, so the identifier should always be 1 or more. And even if it was zero, the key press would have been rejected anyway because the `_GetGenericVkey` lookup would have failed.

I think the actual source of the bug is in the `_GetModifier` method, which sets the `ENHANCED_KEY` flag if the `modifierParam` is greater than 0.

https://github.com/microsoft/terminal/blob/376737e54afc301ced37d5416bc836eadbc611d2/src/terminal/parser/InputStateMachineEngine.cpp#L797

Numeric parameters default to 1, so that should always be greater than 0, but even if this was intended to check for an omitted parameter it doesn't make any sense. The modifiers shouldn't have any affect on the `ENHANCED_KEY` flag should they?

Reply:
Ah, that makes sense. Thank you for reviewing my incorrect change. I've removed the `ENHANCED_KEY` check from `_GetModifier` now and I'll test it later, as I'm hitting another, unrelated bug, which prevents me from testing this change: Win32 input APIs just don't work at  all through WSL for me right now. (Which is also why I submitted this PR late Friday without testing...)
I'm inclined to keep the unrelated change to `_GetGenericKeysModifierState`, as I prefer the range check. It better mirrors how we use the `default:` case in our `switch` statements over these enums and it's somewhat inobvious IMO that this function doesn't need to guard against `!parameters.at(0).has_value()` because it has been implicitly "checked" via `_GetGenericVkey`.

Reply:
But on the other hand a smaller PR would be better... Ugh, I can't decide. Well, let me know if you feel like anything's amiss with the PR!

Reply:
I honestly don't really like either option, but if I had to pick one, I'd be more inclined to leave it as it was, just because it keeps the code shorter. I don't feel strongly about that though.

Longer term I'd like to see if this code could be refactored to make things a bit cleaner, and potentially add support for more keys, but that's not a priority. What you have in the latest commit should fix the `ENHANCED_KEY` issue, which is all that matter for now.

Reply:
Btw, I just use a cmd shell for testing, and trick Window Terminal into disabling the win32 input mode by doing `echo ^[[?9001;9999l` (where `^[` is the <kbd>Esc</kbd> key).